Fix memory leak in error case in psa_crypto (#4131)
ARMmbed/mbedtls: ChangeLog.d/fix_psa_crypto_leak.txt

Bugfix

>    * Fix memory leak that occured in error case in psa_generate_derived_key_internal()

Reviewer: Please use proper English in changelog entries.
```suggestion
   * Fix a memory leak in an error case in psa_generate_derived_key_internal().
```
